src/liballoc/collections/btree/map.rs

```diff
@@ -2781,28 +2779,25 @@ impl<'a, K: 'a, V: 'a> Handle<NodeRef<marker::Mut<'a>, K, V, marker::LeafOrInter
 }
 
 enum UnderflowResult<'a, K, V> {
-    AtRoot(NodeRef<marker::Mut<'a>, K, V, marker::LeafOrInternal>),
-    EmptyParent(NodeRef<marker::Mut<'a>, K, V, marker::Internal>),
+    AtRoot,
     Merged(Handle<NodeRef<marker::Mut<'a>, K, V, marker::Internal>, marker::Edge>, bool, usize),
-    Stole(NodeRef<marker::Mut<'a>, K, V, marker::Internal>, bool),
+    Stole(bool),
```

Comment:
Nice, but now Stole is begging me to be split up into something like StoleFromLeft and StoleFromRight.

Reply:
But then I would have to do the same with `MergedWithRight` and `MergedWithLeft`, and that makes just makes the `match` above more complicated.